### PR TITLE
Add "Support Godot Development" option to the editor's Help menu

### PR DESCRIPTION
**Pull request by Calinou**
Some menu items were also renamed for clarity. (These renames aren't critical, and can be excluded from the `3.3` cherry-pick to avoid breaking translations there.)

## Preview

image

### CODE REVIEW

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -699,11 +699,11 @@ void EditorNode::_notification(int p_what) {
 			p->set_item_icon(p->get_item_index(HELP_SEARCH), gui_base->get_theme_icon("HelpSearch", "EditorIcons"));
 			p->set_item_icon(p->get_item_index(HELP_DOCS), gui_base->get_theme_icon("Instance", "EditorIcons"));
 			p->set_item_icon(p->get_item_index(HELP_QA), gui_base->get_theme_icon("Instance", "EditorIcons"));
-			p->set_item_icon(p->get_item_index(HELP_ABOUT), gui_base->get_theme_icon("Godot", "EditorIcons"));
```
Comment (Calinou) [inline editor/editor_node.cpp:702]: This was a duplicate line, so I removed it.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.1.

**Pull request merged**